src/Services/MarketTradingDataService.cs

```diff
@@ -19,75 +19,127 @@ public MarketTradingDataService(ICandlesHistoryServiceProvider candlesHistorySer
 
-                if (mtCandles.TryGetValue(spotAssetCandles.Key, out var mtAssetCandles))
+                if (candles.Item2.TryGetValue(spotAssetCandles.Key, out var mtAssetCandles))
                 {
                     volume24 += mtAssetCandles.History.Sum(c => c.TradingOppositeVolume);
                 }
 
-                var assetPairData = new AssetPairTradingData(
+                var assetPairDataItem = new AssetPairTradingDataItem<double>(
                     spotAssetCandles.Key,
-                    spotAssetCandles.Value.History.LastOrDefault()?.LastTradePrice ?? 0,
                     volume24);
 
-                result.Add(assetPairData.AssetPair, assetPairData);
+                result.Add(assetPairDataItem.AssetPair, assetPairDataItem);
             }
 
-            foreach (var mtAssetCandles in mtCandles)
+            foreach (var mtAssetCandles in candles.Item2)
             {
                 if (!result.ContainsKey(mtAssetCandles.Key))
                 {
-                    var assetPairData = new AssetPairTradingData(
+                    var assetPairDataItem = new AssetPairTradingDataItem<double>(
                         mtAssetCandles.Key,
-                        mtAssetCandles.Value.History.LastOrDefault()?.LastTradePrice ?? 0,
                         mtAssetCandles.Value.History.Sum(c => c.TradingOppositeVolume));
 
-                    result.Add(assetPairData.AssetPair, assetPairData);
+                    result.Add(assetPairDataItem.AssetPair, assetPairDataItem);
                 }
             }
 
             return result.Values;
         }
 
+        public async Task<IEnumerable<AssetPairTradingDataItem<double>>> TryGetAllPairsLastTradePriceAsync()
+        {
+            var candles = await GetLastDayCandlesAsync(); // Item1 = spot, Item2 = MT
+
+            var result = new Dictionary<string, AssetPairTradingDataItem<double>>();
+
+            foreach (var spotAssetCandles in candles.Item1)
+            {
+                var assetPairDataItem = new AssetPairTradingDataItem<double>(
+                    spotAssetCandles.Key,
+                    spotAssetCandles.Value.History.LastOrDefault()?.Close ?? 0);
+
+                result.Add(assetPairDataItem.AssetPair, assetPairDataItem);
+            }
+
+            foreach (var mtAssetCandles in candles.Item2)
+            {
+                if (!result.ContainsKey(mtAssetCandles.Key))
+                {
+                    var assetPairDataItem = new AssetPairTradingDataItem<double>(
+                        mtAssetCandles.Key,
+                        mtAssetCandles.Value.History.LastOrDefault()?.Close ?? 0);
+
+                    result.Add(assetPairDataItem.AssetPair, assetPairDataItem);
+                }
+            }
+
+            return result.Values;
+        }
+
+        private async
+            Task<Tuple<IReadOnlyDictionary<string, CandlesHistoryResponseModel>,
```

Comment:
Use modern value tuple with clear item names, instead of Tuple class with its Item1 & Item2

Reply:
Probably you even don't need this method, if you implement separate method for the last trade price as noted above